WcaOnRails/app/models/user.rb

```diff
@@ -376,6 +376,10 @@ def can_manage_competition?(competition)
     can_admin_results? || competition.organizers.include?(self) || competition.delegates.include?(self)
   end
 
+  def can_view_competition_results?(competition)
+    can_admin_results?
```

Comment:
Urg, good catch! This actually isn't currently a problem because the `user_can_view_results?` method in competition.rb is the only place that uses this, but you are correct that this is definitely wrong. It's super frustrating that we don't have a place to put these permission rules that allow us to include logic for people who are not logged in.

I'm going to just get rid of this method.